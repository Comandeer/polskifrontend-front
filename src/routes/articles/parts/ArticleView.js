import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import styles from './ArticleView.styl';
import ResponsivePanel from '../../../components/Responsive/ResponsivePanel';
import Link from '../../../components/Link/Link';
import dateFormat from 'dateformat';
import Loader from '../../../components/Indicators/Loader';

const ArticleView = (props) => {
  return (
    <div>
      <Loader isLoading={props.isLoading}>
        <ResponsivePanel className={styles.container}
                         header={props.blogName}
                         description=""
                         image={props.blogIcon}
                         showImage
                         href={props.blogHref}
        >
          <div className={styles.nav}>
            <Link className={styles.back} to="/">
              <i className="icon-left-big">
              </i>
              Strona główna
            </Link>
          </div>
          <div className={styles.item}>
            <h2 className={styles['item__header']}>
              <a href={props.href} target="_blank" rel="nofollow" title={props.title}>{props.title}</a>
            </h2>
            <div className={styles['meta']}>
              <p className={styles['meta__date']}>
                {dateFormat(props.date, 'dd-mm-yyyy')}
              </p>
              <pre className={styles['meta__message']}>
                {props.description}
              </pre>
            </div>
            <div className={styles.more}>
              <h3 className={styles['more__header']}>Chcesz więcej? Przeczytaj w oryginale!</h3>
              <a className={styles['more__link']} href={props.href} target="_blank" rel="nofollow" title={props.title}>
                <i className="icon-link-ext-alt">
                </i>
                Przejdź do artykułu
              </a>
            </div>
          </div>
        </ResponsivePanel>
      </Loader>
    </div>
  );
};

ArticleView.propTypes = {
  blogHref: PropTypes.string.isRequired,
  blogIcon: PropTypes.string.isRequired,
  blogName: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  isLoading: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired
};

export default withStyles(styles)(ArticleView);
