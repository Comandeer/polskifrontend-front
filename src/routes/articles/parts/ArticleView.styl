@import "../../../styles/globals.styl"
@import "../../../styles/variables.styl"
@import "../../../styles/colors.styl"

.container
  padding 0

  .nav
    text-align left
    margin-top $global-margin * 1.5

    .back
      font-weight bold
      display inline-block
      padding $global-padding * 0.75
      background $color-secondary
      color $color-tertiary
      border 3px solid $color-tertiary
      border-radius $global-radius
      text-decoration none
      transition background $animation-time

      @media screen and (min-width: 768px)
        padding $global-padding ($global-padding * 2)

      &:hover
        background $color-tertiary
        color $color-secondary

      &:active
        background $color-secondary
        color $color-tertiary
        transition background 0s

      i
        font-size 20px
        margin-right 10px
        vertical-align top

      a
        font-family $font-header
        font-size 20px

        i
          font-size 20px
          margin-right 10px
          vertical-align top

  .item
    border-left 4px solid $color-tertiary
    padding $global-margin
    margin $global-margin * 1.5 0 $global-margin
    text-align left
    transition border-bottom $animation-time, border-left $animation-time
    position relative
    background $color-secondary
    box-shadow 1px 1px 6px 1px lighten($color-primary, 80%)
    border-radius $global-radius

    &__new
      background $color-tertiary
      color $color-secondary
      padding 5px $global-padding
      display none
      border-radius 5px
      font-weight normal
      font-size 12px
      vertical-align text-bottom
      margin-right $global-padding
      transition background $animation-time

      &--visible
        display inline-block

      &--clicked
        background $color-tertiary

    &__header
      display inline-block
      max-width 100%
      margin 0 0 $global-padding
      text-decoration none
      font-size 20px
      font-weight bold
      font-family $font-header
      color $color-primary
      transition color $animation-time
      white-space nowrap
      overflow hidden
      text-overflow ellipsis
      text-transform none

      a
        color $color-primary
        transition color $animation-time

        &:hover
          color lighten($color-primary, 40%)

    .meta
      padding-top $global-margin
      color lighten($color-primary, 50%)
      display inline

      &__date
        font-size 14px

      &__message
        font-size 1.1em
        line-height 1.6em
        color $color-primary
        padding $global-padding 0 0
        white-space pre-wrap
        font-family $font-text

    .more
      text-align center
      padding $global-margin 0 0
      margin-top $global-margin
      border-top 2px dashed $color-tertiary

      &__header
        text-transform none
        margin 0 0 $global-margin
        font-size 1.4em

      &__link
        font-weight bold
        display inline-block
        padding $global-padding * 0.75
        background $color-secondary
        color $color-tertiary
        border 3px solid $color-tertiary
        border-radius $global-radius
        text-decoration none
        transition background $animation-time

        @media screen and (min-width: 768px)
          padding $global-padding ($global-padding * 2)

        &:hover
          background $color-tertiary
          color $color-secondary

        &:active
          background $color-secondary
          color $color-tertiary
          transition background 0s

        i
          font-size 20px
          margin-right 10px
          vertical-align top

        a
          font-family $font-header
          font-size 20px

          i
            font-size 20px
            margin-right 10px
            vertical-align top
