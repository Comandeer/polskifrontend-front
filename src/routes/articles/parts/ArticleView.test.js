/* eslint-disable */
import React from 'react';
import ArticleView from './ArticleView';
import renderer from 'react-test-renderer';
import StyleProvider from '../../../../test/StyleProvider';

describe('ArticleView component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <StyleProvider insertCss={() => {}}>
          <ArticleView isLoading={false}
                       date={new Date('2019-01-01')}
                       title="Test title"
                       href="/test"
                       blogHref="/test"
                       blogIcon="/test.img"
                       blogName="test-name"
                       description="test-description"
          />
        </StyleProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
