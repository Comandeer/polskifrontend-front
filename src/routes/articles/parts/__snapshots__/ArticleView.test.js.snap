// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleView component renders correctly 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="container container"
    >
      <div
        className="container"
      >
        <h2
          className="container__title"
        >
          <div>
            <a
              className="link"
              href="/test"
              rel="nofollow"
              target="_blank"
            >
              <img
                alt="test-name - ikona"
                className="favicon"
                src="no_image.png"
              />
            </a>
            <span>
              test-name
            </span>
          </div>
        </h2>
        <div
          className="wrapper"
        >
          <p
            className="wrapper__description"
          >
            
          </p>
          <div
            className="nav"
          >
            <a
              className="back"
              href="/"
              onClick={[Function]}
            >
              <i
                className="icon-left-big"
              />
              Strona główna
            </a>
          </div>
          <div
            className="item"
          >
            <h2
              className="item__header"
            >
              <a
                href="/test"
                rel="nofollow"
                target="_blank"
                title="Test title"
              >
                Test title
              </a>
            </h2>
            <div
              className="meta"
            >
              <p
                className="meta__date"
              >
                01-01-2019
              </p>
              <pre
                className="meta__message"
              >
                test-description
              </pre>
            </div>
            <div
              className="more"
            >
              <h3
                className="more__header"
              >
                Chcesz więcej? Przeczytaj w oryginale!
              </h3>
              <a
                className="more__link"
                href="/test"
                rel="nofollow"
                target="_blank"
                title="Test title"
              >
                <i
                  className="icon-link-ext-alt"
                />
                Przejdź do artykułu
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
